cunumeric/array.py

```diff
@@ -2781,7 +2778,7 @@ def prod(
             UnaryRedCode.PROD,
             self_array,
             axis=axis,
-            dst=out,
+            out=out,
             keepdims=keepdims,
             initial=initial,
             where=where,
```

Comment:
The `dtype` argument of `prod` is not passed through.

Reply:
dbb22229a813684621c0d0228f9171120e576b0b